### ISSUE TITLE
New `Unexpected rank, expected 3, got: 2`

### ISSUE THREAD

**Issue by Jeadie** [EricLBuehler/mistral.rs#319]
**Describe the bug**
Running a llama model variant via GGUF. Had no issues, until an upgrade of versions (see below). 

#### General config
```
general.architecture: llama
general.file_type: 7
general.name: models
general.quantization_version: 2
llama.attention.head_count: 32
llama.attention.head_count_kv: 32
llama.attention.layer_norm_rms_epsilon: 0.00001
llama.block_count: 32
llama.context_length: 16384
llama.embedding_length: 4096
llama.feed_forward_length: 11008
llama.rope.dimension_count: 128
llama.rope.freq_base: 1000000
```

Sample Rust code (link attached below). 
```rust
let pipeline = GGUFLoaderBuilder::new(
            GGUFSpecificConfig::default(),
            None,
            Some(tokenizer.to_string_lossy().to_string()),
            Some("motherduckdb/DuckDB-NSQL-7B-v0.1-GGUF".to_string()),
            "quantized_model_id".to_string(),
            model_weights.to_string_lossy().to_string(),
        )
        .build()
        .load_model_from_path(
            &paths,
            None,
            &candle_core_rs::Device::Cpu,
            false,
            DeviceMapMetadata::dummy(),
            None,
        )
        .map_err(|e| NqlError::FailedToLoadModel { source: e.into() })?;

        let (tx, rx) = channel(10_000);
        Ok(Self {
            pipeline: MistralRsBuilder::new(
                pipeline,
                SchedulerMethod::Fixed(5.try_into().map_err(|_| NqlError::FailedToLoadModel {
                    source: "couldn't create schedule method".into(),
                })?),
            )
            .build(),
            tx,
            rx,
        })
```
 https://github.com/spiceai/spiceai/blob/jeadie/24-05-09/llms/crates/llms/src/nql/mistral.rs

Error: `unexpected rank, expected: 3, got: 2 ([1, 96])`

**Latest commit**
Commit: f09b4e5beddba91fcb9430beb8bf58b86b57028a
Prior commit without bug: f9b85c92767e8c644ee921606ddb14433014077c

**Comment by Jeadie** [comment]
Given the `CompletionResponse`, I suspect it is an issue at the first layer (since `prompt_tokens=96`)
```
CompletionResponse {
    id: "0",
    choices: [
        CompletionChoice {
            finish_reason: "error",
            index: 0,
            text: "",
            logprobs: None,
        },
    ],
    created: 1715826020,
    model: "motherduckdb/DuckDB-NSQL-7B-v0.1-GGUF",
    system_fingerprint: "local",
    object: "text_completion",
    usage: Usage {
        completion_tokens: 0,
        prompt_tokens: 96,
        total_tokens: 96,
        avg_tok_per_sec: 48000.0,
        avg_prompt_tok_per_sec: inf,
        avg_compl_tok_per_sec: NaN,
        total_time_sec: 0.002,
        total_prompt_time_sec: 0.0,
        total_completion_time_sec: 0.0,
    },
}
```

**Comment by EricLBuehler** [comment]
@Jeadie, was this fixed by #317/#320?

**Comment by Jeadie** [comment]
Fixed
